### ISSUE TITLE
Unexpected with _.orderBy

### ISSUE THREAD

**Issue by slorber** [lodash/lodash#2011]
Hi,

This may not be a bug and I may be using the function the wrong way, but I was surprised these 2 calls where actually producing different results, and did not find the documentation really helpful to understand what was happening (about the [iteratees] stuff)

```javascript
var users = [
  { 'user': 'fred',   'age': 48 },
  { 'user': 'barney', 'age': 34 },
  { 'user': 'fred',   'age': 42 },
  { 'user': 'barney', 'age': 36 },
  { 'user': 'barney', 'age': 4 },
  { 'user': 'barney', 'age': 3 }
];

console.log("users",users);

var result = _.orderBy(users, ['user','age'], ['asc', 'desc']);
console.log("result",result);

var result2 = _.orderBy(users, function(item) { return [item.user,item.age]}, ['asc', 'desc']);
console.log("result2",result2);
```

https://jsfiddle.net/dwpcz0dx/3/

As you can notice, the first and second results are not ordered the same way because of the "age" property. This seems strange and counterintuitive to me. It is like the age property is being handled as a number in the first case, and as a string in the 2nd case (thus producing an order like "4" > "36" > "3").

I have the same problem when doing:

```javascript
console.log(
	_.orderBy([1,11,5,4,36,50,3])
);
console.log(
	_.orderBy([1,11,5,4,36,50,3],_.identity)
);
console.log(
	_.orderBy([1,11,5,4,36,50,3],function(item) { return [item];})
);
```

The 3rd one is actually not returning the same order while I expected it would.

----------------------------------------------------------------------

My real usecase: I want to sort a list of objects of shape {a: string, b:number}. I want to sort by 4 criterias, a,b,c and d, by asc/desc order. a and b criterias are object properties, c (string) and d (number) are computed values that I don't want to store in the object shape.
I don't understand how to do that with lodash even if looking closely at the documentation.

**Comment by slorber** [comment]
Ohhh sorry I've just figured it out:

```javascript
      const sortedTags = _.orderBy(
        tags,
        [tag => _.includes(selectedTags,tag.value),"count","value"],
        ["desc","desc","asc"]
      );
```

**Comment by jdalton** [comment]
Remember the second param, `iteratees`, is an array of iteratees or their shorthand.
An `iteratees` of `['user', 'age']` is not equiv to `function(item) {return [item.user,item.age]}` it's equiv to
`[function(item){return item.user}, function(item){return item.age}]`

**Comment by slorber** [comment]
yes @jdalton I've figuered it out, but it was not clear in doc and searching for "iteratee" did not really help to understand what it is supposed to be. I see it used in other functions but it did not really helped me.

I understood better that I had to pass an array with the signature `[iteratees=[_.identity]] (Function[]|Object[]|string[]):` but it was kind of confusing because it let us think we can't melt functions and strings into the same array (like I did). Also was confused by the fact that not providing an array, but actually providing a function, seemed to almost work while the signature does not say we can pass a function (but I've seen that in some SO answer I think)

I already know iteratees in FRP and it does not look like the lodash iteratee: https://en.wikipedia.org/wiki/Iteratee
